src/Util.cpp

```diff
@@ -1346,10 +1352,29 @@ traverse(const std::string& path, const TraverseVisitor& visitor)
     visitor(path, true);
   } else if (errno == ENOTDIR) {
     visitor(path, false);
+#else // If not available, use C++17 std::filesystem implementation.
+  if (std::filesystem::is_directory(path)) {
+    for (auto&& p : std::filesystem::directory_iterator(path)) {
+      std::string entry = p.path().string();
+
+      if (std::filesystem::is_directory(p)) {
+        traverse(entry, visitor);
+      } else if (std::filesystem::is_regular_file(p)) {
```

Comment:
Why is this not simply else? Is it the same reason you don't use `std::filesystem::recusive_directory_iterator`?

Reply:
What if it's not a directory and not a type of file that is a regular file? How should that be handled?

I tried the recursive one, but unit tests were failing because the order was not what it was expecting.

Reply:
I've cleaned up the code a bit, do you like this version better?